src/sema/expression.rs

```diff
@@ -7885,6 +7904,34 @@ pub fn call_expr(
     Ok(expr)
 }
 
+/// is it an (new C).value(1).gas(2)(1, 2, 3) style constructor (not supported)
+fn old_style_constructor_arguments(expr: &pt::Expression, diagnostics: &mut Diagnostics) -> bool {
+    match expr.remove_parenthesis() {
+        pt::Expression::FunctionCall(func_loc, ty, _) => {
+            if let pt::Expression::MemberAccess(_, ty, call_arg) = ty.as_ref() {
+                if old_style_constructor_arguments(ty, diagnostics) {
+                    // location should be the identifier and the arguments
+                    let mut loc = call_arg.loc;
+                    if let pt::Loc::File(_, _, end) = &mut loc {
+                        *end = func_loc.end();
+                    }
+                    diagnostics.push(Diagnostic::error(
+                        loc,
+                        format!("old style call argument using '.{}(...)' is not supported, use '{{{}: ...}}' instead", call_arg.name, call_arg.name)
```

Comment:
Nit: "Old style call arguments" sounds a bit.. lax? What about `deprecated` or omitting it all together?

```suggestion
                        format!("deprecated call argument syntax '.{}(...)' is not supported, use '{{{}: ...}}' instead", call_arg.name, call_arg.name)
```